cmd/skopeo/completions.go

```diff
@@ -4,10 +4,47 @@ import (
 	"github.com/containers/image/v5/tarball"
 	"github.com/containers/image/v5/transports"
 	"github.com/spf13/cobra"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
 )
 
-// autocompleteSupportedTransports list all supported transports with the colon suffix.
-func autocompleteSupportedTransports(cmd *cobra.Command, args []string, toComplete string) ([]cobra.Completion, cobra.ShellCompDirective) {
+func autocompleteTransports(cmd *cobra.Command, args []string, toComplete string) ([]cobra.Completion, cobra.ShellCompDirective) {
+	directive := cobra.ShellCompDirectiveNoSpace
+	// We still don't have a transport
+	if !strings.Contains(toComplete, ":") {
```

Comment:
Overall this would likely be much better written with `strings.Cut()`

That way you can easily check if a colon was set and the you have the clear prefix and can match that without the colon below.

Reply:
Yeah, this works much better. Updated to do this